rdr_service/dao/bq_participant_summary_dao.py

```diff
@@ -419,77 +430,164 @@ def _calculate_enrollment_status(self, p_id, ro_session, ro_summary):
+            response_date = consent['consent_date']
             if consent['consent'] == 'ConsentPII':
                 study_consent = True
-            if consent['consent'] == 'EHRConsentPII_ConsentPermission':
-                consents['EHRConsent'] = (consent['consent_value'], consent['consent_date'])
-            if consent['consent'] == 'DVEHRSharing_AreYouInterested':
-                consents['DVEHRConsent'] = (consent['consent_value'], consent['consent_date'])
+                study_consent_date = min(study_consent_date, response_date)
+            elif consent['consent'] == EHR_CONSENT_QUESTION_CODE:
+                consents['EHRConsent'] = (response_value, response_date)
+                had_ehr_consent = had_ehr_consent or response_value == CONSENT_PERMISSION_YES_CODE
+            elif consent['consent'] == DVEHR_SHARING_QUESTION_CODE:
+                consents['DVEHRConsent'] = (response_value, response_date)
+                had_ehr_consent = had_ehr_consent or response_value == DVEHRSHARING_CONSENT_CODE_YES
+            elif consent['consent'] == GROR_CONSENT_QUESTION_CODE:
+                consents['GRORConsent'] = (response_value, response_date)
+                had_gror_consent = had_gror_consent or response_value == CONSENT_GROR_YES_CODE
 
         if 'EHRConsent' in consents and 'DVEHRConsent' in consents:
-            if consents['DVEHRConsent'] == 'DVEHRSharing_Yes' and consents['EHRConsent'][0] != 'ConsentPermission_No':
+            if consents['DVEHRConsent'] == DVEHRSHARING_CONSENT_CODE_YES\
+                    and consents['EHRConsent'][0] != CONSENT_PERMISSION_NO_CODE:
                 ehr_consent = True
-            if consents['EHRConsent'][0] == 'ConsentPermission_Yes':
+            if consents['EHRConsent'][0] == CONSENT_PERMISSION_YES_CODE:
                 ehr_consent = True
         elif 'EHRConsent' in consents:
-            if consents['EHRConsent'][0] == 'ConsentPermission_Yes':
+            if consents['EHRConsent'][0] == CONSENT_PERMISSION_YES_CODE:
                 ehr_consent = True
         elif 'DVEHRConsent' in consents:
             if consents['DVEHRConsent'][0] == 'DVEHRSharing_Yes':
                 ehr_consent = True
 
+        if 'GRORConsent' in consents:
+            gror_answer = consents['GRORConsent'][0]
+            gror_consent = gror_answer == CONSENT_GROR_YES_CODE
+
         # check physical measurements
+        physical_measurements_date = datetime.datetime.min
         if 'pm' in ro_summary:
             for pm in ro_summary['pm']:
                 if pm['pm_status_id'] == int(PhysicalMeasurementsStatus.COMPLETED) or \
-                    (pm['pm_finalized'] and pm['pm_status_id'] != int(PhysicalMeasurementsStatus.CANCELLED)):
+                        (pm['pm_finalized'] and pm['pm_status_id'] != int(PhysicalMeasurementsStatus.CANCELLED)):
                     pm_complete = True
+                    physical_measurements_date = min(physical_measurements_date, pm['pm_finalized'])
```

Comment:
Per offline conversation:  Seems this will never result in an actual date from the participant history, physical_measurements_date will always retain the original datetime.datetime.min assigned value from line 473?   Is that the desired logic here?

Reply:
Nope, updating it now to find the earliest physical measurements time.